group-coordinator/src/test/java/org/apache/kafka/coordinator/group/OffsetMetadataManagerTest.java

```diff
@@ -70,47 +70,66 @@
 import java.util.OptionalLong;
 
 import static org.apache.kafka.common.requests.OffsetFetchResponse.INVALID_OFFSET;
+import static org.apache.kafka.coordinator.group.OffsetMetadataManager.OffsetExpirationCondition.DEFAULT_OFFSET_EXPIRATION_CONDITION;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.junit.jupiter.api.Assertions.assertNull;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
 public class OffsetMetadataManagerTest {
     static class OffsetMetadataManagerTestContext {
         public static class Builder {
-            final private MockTime time = new MockTime();
-            final private MockCoordinatorTimer<Void, Record> timer = new MockCoordinatorTimer<>(time);
-            final private LogContext logContext = new LogContext();
-            final private SnapshotRegistry snapshotRegistry = new SnapshotRegistry(logContext);
+            private final MockTime time = new MockTime();
+            private final MockCoordinatorTimer<Void, Record> timer = new MockCoordinatorTimer<>(time);
+            private final LogContext logContext = new LogContext();
+            private final SnapshotRegistry snapshotRegistry = new SnapshotRegistry(logContext);
+            private GroupMetadataManager groupMetadataManager = null;
             private MetadataImage metadataImage = null;
-            private int offsetMetadataMaxSize = 4096;
+            private GroupCoordinatorConfig config = null;
 
             Builder withOffsetMetadataMaxSize(int offsetMetadataMaxSize) {
-                this.offsetMetadataMaxSize = offsetMetadataMaxSize;
+                config = GroupCoordinatorConfigTest.createGroupCoordinatorConfig(offsetMetadataMaxSize, 60000L, 24 * 60 * 1000);
+                return this;
+            }
+
+            Builder withOffsetsRetentionMs(long offsetsRetentionMs) {
+                config = GroupCoordinatorConfigTest.createGroupCoordinatorConfig(4096, 60000L, offsetsRetentionMs);
```

Comment:
I am not a big fan of this because it won't scale if we add another params. It seems to be that we should have a config builder in the test package with sane default for tests. Then we could pass the config object here or instantiate the default one. What do you think?

I am fine with doing this as a follow-up though.